pkg/kubelet/volume/volume_manager.go

```diff
@@ -185,10 +187,17 @@ type volumeManager struct {
 	// desiredStateOfWorldPopulator runs an asynchronous periodic loop to
 	// populate the desiredStateOfWorld using the kubelet PodManager.
 	desiredStateOfWorldPopulator populator.DesiredStateOfWorldPopulator
+
+	// mountedVolulmeToClean is the list of volumes already mounted before
+	// kubelet starts. They need to be cleaned up before reconciler runs
+	mountedVolumeToClean []operationexecutor.MountedVolume
 }
 
 func (vm *volumeManager) Run(stopCh <-chan struct{}) {
 	defer runtime.HandleCrash()
+	glog.Infof("Clean up already mounted volumes before Kubelet starts")
+	vm.cleanUpMountedVolumes()
+
```

Comment:
Yes, but you only perform this cleanup once. For example, if there is problem accessing one directory for whatever transient error, the cleanup task fails, but the code never retries. The regular cleanup routine will not deal with these orphaned volumes as we all know.